util/src/main/java/tech/pegasys/teku/util/exceptions/ExceptionUtil.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.util.exceptions;
+
+import java.util.Optional;
+
+public class ExceptionUtil {
+
+  @SuppressWarnings("unchecked")
+  public static <T extends Throwable> Optional<T> getCause(
+      final Throwable err, Class<T> targetType) {
+    Throwable current = err;
+    while (current != null) {
+      if (targetType.isInstance(current)) {
+        return Optional.of((T) current);
+      }
+      current = current.getCause();
+    }
+    return Optional.empty();
+  }
```

Comment:
It may be worth using Guava's `Throwables.getCausalChain` so we benefit from all the details they have in iterating exception causes like loop detection.  I think it would be something like:
```suggestion
  public static <T extends Throwable> Optional<T> getCause(
      final Throwable err, Class<T> targetType) {
    return Throwables.getCausalChain(err).filter(cause -> targetType.isInstance(cause)).findFirst();
  }
```

Reply:
Used apache commons `ExceptionUtils.getThrowableList()` because the guava util throws if there's a loop.